From 7a735e3018eff5df906c64bb242db236320a9583 Mon Sep 17 00:00:00 2001
From: Phil Sutter <psutter@redhat.com>
Date: Fri, 17 Mar 2017 13:25:13 +0100
Subject: [PATCH] tc: flower: Update documentation to indicate ARP takes IPv4
 prefixes

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1422629
Upstream Status: iproute2.git commit c7ec052bb8682

commit c7ec052bb8682a9889d1e71e2cb2ec780c9d5e26
Author: Simon Horman <simon.horman@netronome.com>
Date:   Wed Feb 8 13:01:04 2017 +0100

    tc: flower: Update documentation to indicate ARP takes IPv4 prefixes

    Unlike other PREFIXes documented in the usage for tc flower, which accept
    both IPv4 and IPv6 prefixes, arp_sip and arp_tip only accepts IPv4
    prefixes.

    Signed-off-by: Simon Horman <simon.horman@netronome.com>
---
 man/man8/tc-flower.8 | 8 ++++----
 tc/f_flower.c        | 4 ++--
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/man/man8/tc-flower.8 b/man/man8/tc-flower.8
index 2bdd2ef..b1bef8b 100644
--- a/man/man8/tc-flower.8
+++ b/man/man8/tc-flower.8
@@ -36,7 +36,7 @@ flower \- flow based traffic control filter
 .BR dst_port " | " src_port " } "
 .IR port_number " } | { "
 .BR arp_tip " | " arp_sip " } "
-.IR PREFIX " | "
+.IR IPV4_PREFIX " | "
 .BR arp_op " { " request " | " reply " | "
 .IR OP " } | { "
 .BR arp_tha " | " arp_sha " } "
@@ -139,11 +139,11 @@ Match on ICMP type or code. Only available for
 .BR ip_proto " values " icmp  " and " icmpv6
 which have to be specified in beforehand.
 .TP
-.BI arp_tip " PREFIX"
+.BI arp_tip " IPV4_PREFIX"
 .TQ
-.BI arp_sip " PREFIX"
+.BI arp_sip " IPV4_PREFIX"
 Match on ARP or RARP sender or target IP address.
-.I PREFIX
+.I IPV4_PREFIX
 must be a valid IPv4 address optionally followed by a slash and the prefix
 length. If the prefix is missing, \fBtc\fR assumes a full-length host
 match.
diff --git a/tc/f_flower.c b/tc/f_flower.c
index b4bf886..a2b2132 100644
--- a/tc/f_flower.c
+++ b/tc/f_flower.c
@@ -59,8 +59,8 @@ static void explain(void)
 		"                       src_port PORT-NUMBER |\n"
 		"                       type ICMP-TYPE |\n"
 		"                       code ICMP-CODE |\n"
-		"                       arp_tip PREFIX |\n"
-		"                       arp_sip PREFIX |\n"
+		"                       arp_tip IPV4-PREFIX |\n"
+		"                       arp_sip IPV4-PREFIX |\n"
 		"                       arp_op [ request | reply | OP ] |\n"
 		"                       arp_tha MASKED-LLADDR |\n"
 		"                       arp_sha MASKED-LLADDR |\n"
-- 
1.8.3.1

